src/services/NotificationService.ts

```diff
@@ -0,0 +1,47 @@
+// @ts-ignore
+import {NotifyClient} from "notifications-node-client";
+import {Service} from "../models/injector/ServiceDecorator";
+import {NotifyClientMock} from "../../tests/models/NotifyClientMock";
+import {TEMPLATEIDS} from "../assets/enum";
+import { AWSError } from "aws-sdk";
+
+/**
+ * Service class for Certificate Notifications
+ */
+@Service()
+class NotificationService {
+    private readonly notifyClient: NotifyClient | NotifyClientMock;
+
+    constructor(notifyClient: NotifyClient | NotifyClientMock) {
+        this.notifyClient = notifyClient;
+    }
+
+    /**
+     * Sending email with the certificate according to the given params
+     * @param params - personalization details,email and certificate
+     */
+    public sendNotification(params: any, emails: string[]) {
+        const emailDetails = {
+            personalisation: params
+        };
+        const sendEmailPromise = [];
+        for (const email of emails) {
+            const sendEmail = this.notifyClient.sendEmail(TEMPLATEIDS.ATFReportTemplate, email, emailDetails)
+                .then((response: any) => {
+                    return response;
+                })
+                .catch((err: any) => {
+                    console.error(err);
```

Comment:
same as above

Reply:
I suspect this was copy and paste from other services. We will need to clean those up too once an approach has been decided

Reply:
what approach @KHodderDVSA ? General practice, never gulp exceptions (unless there is a strong business need to catch some custom error), always log and throw them back. Since we are using queues, more of a reason to return back error to the queue so that it appears back for retries.

Reply:
Agreed. These errors are caught by a pyramid of promise-catches, so need to throw errors at every level until we properly throw a response back to the Lambda container.

Reply:
in my old days if i had done this it would be log errors where they occur (service classes in this case) and throw them back so that it propagate backs to the the function aggregating the results (in our case functions)
What do you suggest @KHodderDVSA ?